index.js

```diff
@@ -450,6 +471,83 @@ methods.concat('all').forEach(function (method) {
   }
 })
 
+/**
+ * Add a route to the map with the given path and methods.
+ * @param {Map} routeMap
+ * @param {string} path
+ * @param {Array} methods
+ * @private
+ */
+function addRouteToMap (routeMap, path, methods) {
+  if (routeMap.has(path)) {
+    const existingMethods = routeMap.get(path)
+    for (const method of methods) {
+      if (!existingMethods.includes(method)) {
+        existingMethods.push(method)
+      }
+    }
+  } else {
+    routeMap.set(path, [...methods])
+  }
+}
+
+/**
+ * Normalize a path by removing trailing slashes.
+ * @param {string} path
+ * @return {string} normalized path
+ * @private
+ */
+function normalizePath (path) {
+  if (typeof path !== 'string') {
+    return path
+  }
+
+  if (path.endsWith('/') && path.length > 1) {
+    return path.slice(0, -1)
+  }
+
+  return path
+}
+
+/**
+ * Collect routes from a router stack recursively.
+ *
+ * @param {Array} stack - The router stack to collect routes from
+ * @param {string} prefix - The path prefix to prepend to routes
+ * @param {Map} routeMap - The map to store collected routes
+ * @private
+ */
+function collectRoutes (stack, prefix, routeMap) {
+  for (const layer of stack) {
+    // for routes without a .use
+    if (layer.pathPatterns && layer.route) {
+      const methods = Object.keys(layer.route.methods).map((method) => method.toUpperCase())
+      if (Array.isArray(layer.pathPatterns)) {
+        for (const pathPattern of layer.pathPatterns) {
+          const fullPath = prefix + pathPattern
```

Comment:
```suggestion
          const fullPath = prefix === '/' ? pathPattern : prefix + pathPattern
```
to fix https://github.com/pillarjs/router/pull/174/files#r2237920545

Reply:
thanks, good catch